### ISSUE TITLE
Support mozilla-central issues ingestion in bot

### ISSUE THREAD

**Issue by La0** [mozilla/code-review#1436]
We need to make [ingest_autoland](https://github.com/mozilla/code-review/blob/master/bot/code_review_bot/workflow.py#L107) more generic in order to support ingestion from mozilla-central (trigerred by #1435 )

**Comment by La0** [comment]
You'll need to support another environment variable `MOZILLA_CENTRAL_TASK_GROUP_ID`, that will trigger the new generic method, with a `Revision` populated in the same way (only the repo's url change in the end)

**Comment by La0** [comment]
We need more work here to identify the phabricator revision from Mozilla-Central. It's required to be able to publish there. We'll simply use the top-most MC revision of the push available on Phabricator.

This can be done by using libmozdata `search_diff` method and iterating over results until you find the commit in the `refs`.

Here is an example:

```
   {
      "id": 672351,
      "type": "DIFF",
      "phid": "PHID-DIFF-lsoffygbolkygiy7g5tr",
      "fields": {
        "revisionPHID": "PHID-DREV-lnyj4zkhd5zbrz7u6cwd",
        "authorPHID": "PHID-USER-dat4prqkqf43erltdjp3",
        "repositoryPHID": "PHID-REPO-saax4qdxlbbhahhp2kg5",
        "refs": [
          {
            "type": "branch",
            "name": "default"
          },
          {
            "type": "base",
            "identifier": "5b25444c8911a4c873b53b6be9f47158831a86b0"
          }
        ],
        "dateCreated": 1674557834,
        "dateModified": 1674557837,
        "policy": {
          "view": "public"
        }
      },
      "attachments": {}
    }
```

Ideally, you should provide the repository PHID to `search_diffs`:
1. call libmozdata `list_repositories`
2. find the one for Mozilla-central and get its PHID
3. provide that phid to `search_diffs` to only get relevant revisions

Please limit the `search_diffs` pagination to avoid loading revisions for a really long time (the bot should crash if the revision is not immediately found).

I suggest implementing a new method `Revision.from_mozilla_central_group` that implement this workflow.

**Comment by vrigal** [comment]
> phabricator.py: Conduit API error ERR-CONDUIT-CORE : Constraint "repositoryPHIDs" is not a valid constraint for this query

@La0 are you sure we can filter by repository ? I could not found more documentation about conduit API.

Also, the phabricator helper seems to list only the 100 first results from the API.

**Comment by La0** [comment]
You're right, it's not available there as a constraint (it's available on revisions :cry: )
